tests/api/forgot-password.test.ts

```diff
@@ -0,0 +1,280 @@
+          email: "walter@gmail.com",
+        }),
+      ).rejects.toThrow("No user found with given email.");
+
+      expect(mockEmails.setApiKey).not.toBeCalled();
+      expect(mockEmails.send).not.toBeCalled();
+    });
+
+    test("Valid user. No pending password reset request.", async () => {
+      await cleanupPasswordResetRequest("walter@gmail.com");
+      await createAccount("walter@gmail.com");
+
+      const response = await $trpcCaller.sendForgotPasswordEmail({
+        email: "walter@gmail.com",
+      });
+
+      expect(mockEmails.setApiKey).toBeCalled();
+      expect(mockEmails.send).toBeCalled();
+
+      expect(response.message).toBe(
+        "Password reset email sent to walter@gmail.com.",
+      );
+
+      const user = await prisma.user.findUnique({
+        where: {
+          email: "walter@gmail.com",
+        },
+        include: {
+          passwordResetReq: true,
+        },
+      });
+      if (!user) {
+        throw new Error("User should not be null.");
+      }
+
+      const passwordResetReq = await prisma.passwordResetReq.findMany({
+        where: {
+          user: user,
+        },
+      });
+      if (!passwordResetReq[0]?.expiresAt) {
+        throw new Error("There should be a password reset request.");
+      }
+
+      expect(passwordResetReq.length).toBe(1);
+      expect(passwordResetReq[0]?.expiresAt > new Date()).toBe(true);
+      expect(user.passwordResetReq?.passwordResetId).toBe(
+        passwordResetReq[0].passwordResetId,
+      );
+
+      await cleanupAccount("walter@gmail.com");
+      await cleanupPasswordResetRequest("walter@gmail.com");
+    });
+
+    test("Valid user. Pending password reset request.", async () => {
+      await cleanupPasswordResetRequest("walter@gmail.com");
+      await createAccount("walter@gmail.com");
+      await createPasswordResetRequest(
+        "walter@gmail.com",
+        new Date(Date.now() + 60_000 * 100000),
+      );
+
+      const user = await prisma.user.findUnique({
+        where: {
+          email: "walter@gmail.com",
+        },
+      });
+      if (!user) {
+        throw new Error("User should not be null.");
+      }
+      let passwordResetReqs = await prisma.passwordResetReq.findMany({
+        where: {
+          user: user,
+        },
+      });
+      expect(passwordResetReqs.length).toBe(1);
+
+      const response = await $trpcCaller.sendForgotPasswordEmail({
+        email: "walter@gmail.com",
+      });
+
+      expect(mockEmails.setApiKey).toBeCalled();
+      expect(mockEmails.send).toBeCalled();
+
+      expect(response.message).toBe(
+        "Password reset email sent to walter@gmail.com.",
+      );
+
+      const userUpdated = await prisma.user.findUnique({
+        where: {
+          email: "walter@gmail.com",
+        },
+        include: {
+          passwordResetReq: true,
+        },
+      });
+      if (!userUpdated) {
+        throw new Error("User should not be null.");
```

Comment:
It's not good practice to throw an error in tests, rework the logic to fail an `expect()` check instead